src/rocks_compaction_scheduler.h

```diff
@@ -28,29 +28,37 @@
 
 #pragma once
 
-#include <atomic>
-#include <set>
-#include <unordered_map>
-#include <memory>
+#include <deque>
 #include <string>
-#include <list>
+
+#include <boost/thread/thread.hpp>
 
 #include <rocksdb/db.h>
 #include <rocksdb/slice.h>
 
+#include "mongo/base/status.h"
 #include "mongo/stdx/mutex.h"
 #include "mongo/util/timer.h"
 
 namespace mongo {
 
+    class RocksCompactionOperation;
+
     class RocksCompactionScheduler {
     public:
-        RocksCompactionScheduler(rocksdb::DB* db) : _db(db) { _timer.reset(); }
+        RocksCompactionScheduler(rocksdb::DB* db);
+        ~RocksCompactionScheduler();
 
         static int getSkippedDeletionsThreshold() { return kSkippedDeletionsThreshold; }
 
         void reportSkippedDeletionsAboveThreshold(const std::string& prefix);
 
+        // schedule compact range operation for execution in _compactionThread
+        Status compactRange(const rocksdb::Slice* begin, const rocksdb::Slice* end);
```

Comment:
Let's accept and keep `std::string` instead of `rocksdb::Slice` for all the methods. I don't like that `rocksdb::Slice` doesn't have underlying data storage beneath it so we always need to be cautious how to handle it properly. `std::string` is easier and safer.